crates/driver/src/domain/competition/solution/trade.rs

```diff
@@ -76,10 +80,43 @@ impl Fulfillment {
         }
     }
 
+    pub fn calculate_custom_prices(
```

Comment:
nit: maybe add a fn comment, where it should be explained that here custom prices are built using the uniform clearing prices and the fee, so that custom prices represent the actual traded amounts as seen from the user perspective (amounts flowing in and out of their wallet)